dev/InfoBar/InfoBarAutomationPeer.cpp

```diff
@@ -0,0 +1,68 @@
+﻿#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "InfoBarAutomationPeer.h"
+#include <UIAutomationCore.h>
+#include <UIAutomationCoreApi.h>
+
+#include "InfoBarAutomationPeer.properties.cpp"
+
+InfoBarAutomationPeer::InfoBarAutomationPeer(winrt::InfoBar const& owner) : ReferenceTracker(owner)
+{
+}
+
+winrt::AutomationControlType InfoBarAutomationPeer::GetAutomationControlTypeCore()
+{
+    return winrt::AutomationControlType::StatusBar;
+}
+
+winrt::hstring InfoBarAutomationPeer::GetClassNameCore()
+{
+    return winrt::hstring_name_of<winrt::InfoBar>();
+}
+
+void InfoBarAutomationPeer::RaiseOpenedEvent(winrt::InfoBarSeverity severity, wstring_view const& displayString)
+{
+    if (winrt::IAutomationPeer7 automationPeer7 = *this)
+    {
+        automationPeer7.RaiseNotificationEvent(
+            winrt::Automation::Peers::AutomationNotificationKind::Other,
+            GetProcessingForSeverity(severity),
+            displayString,
+            L"InfoBarOpenedActivityId");
+    }
+}
+
+void InfoBarAutomationPeer::RaiseClosedEvent(winrt::InfoBarSeverity severity, wstring_view const& displayString)
```

Comment:
>RaiseClosedEvent [](start = 28, length = 16)

Infobar doesn't implement any automation patterns?

Reply:
I didn't find any patterns to implement.